kernel/malloc.c

```diff
@@ -19,7 +19,7 @@
 #define LACKS_STDLIB_H
 #define LACKS_SCHED_H
 #define LACKS_TIME_H
-#define ABORT_ON_ASSERT_FAILURE 1
+#define ABORT_ON_ASSERT_FAILURE 0
```

Comment:
Good catch. It's not a typo, but the DEBUG business in malloc() is slightly confusing. I've pushed some changes to make our position clearer, and fix compliation with DEBUG=1 should anyone need it.